---
title: "Cardiogenic shock RNA-seq (GSE131411) processing"
author: "Patrick Maclean"
date: "2023-07-03"
output: 
  html_document: 
    toc: yes
---

# Introduction
This is a repeat analysis of:
https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE131411
Whole blood transcriptome analysis of Septic shock (SS) and Cardiogenic shock (CS) patients over one week of ICU stay (ShockOmics study)

Dataset:
11 patients with cardiogenic shock and 21 with septic shock, with samples collected at 16 hours, 48 hours and 7 days. Only patients still in hospital at 7 days were included. 

Main questions:
1. Does SRS (categorical and quantitative) associate with mortality or clinical severity in cardiogenic shock?
1-a. If so, is there the expected relationship with the estimated IL1R2+ neutrophil proportion (estimated using CIBERSORTx and the Kwok whole-blood sepsis scRNA-seq dataset)

2. Are there differences in the neutrophil-lymphocyte ratio across CS/SS (as estimated with CIBERSORTx)? Was this overlooked in their analysis?

3. Controlling for the NLR (ie including it in the differential expression model), do we see any differences in the differential gene expression analysis, compared to theirs where they found very little difference across SS/CS? 

Hypothesis: In this dataset, the NLR starts high in both groups, but crucially it is much higher in the septic shock than cardiogenic shock groups. They don't control for this difference in their model, so they fail to identify any differences between the cardiogenic and septic shock groups. They are almost certainly there but confounded by the differences in NLR. Using a model incorporating the NLR as a confounder should demonstrate this. Unfortunately there is no cell count data available with the paper, but we should be able to estimate this with CIBERSORTx. 


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Setup
## Load packages
```{r echo = T, results = 'hide', message = FALSE}
library(clusterProfiler)
library(ggplot2)
library(ggrepel)
library(pheatmap)
library(biomaRt)
library(SepstratifieR)
library(janitor)
library(DESeq2)
library(edgeR)
library(org.Hs.eg.db)
library(table1)
library(patchwork)
library(msigdbr)
library(tidyverse)
library(lmerSeq)
library(viridis)
library(ggpubr)
library(ggtext)
library(ggforce)
library(stringi)
```

## Set global plotting theme
Use thematic package to set a custom colour theme for this analysis
```{r results = 'hide', message = FALSE}
library(scales)
library(thematic)
colour_sequence <- c("#35274A", "#F2300F","#0B775E","#E1BD6D")

thematic_on(
  bg = "auto", fg = "auto", accent = "auto", font = NA,
  sequential = colour_sequence, qualitative = okabe_ito()
)
show_col(colour_sequence)
```

# Load data
## GEO file
Load GEO file with raw counts per transcript - have not repeated the alignment from .fastqc files for this analysis
These are labeled with Ensembl type identifiers (like ENSG00000000003) - we need these to run sepstratifier, but mostly we'll want HGNC identifiers (eg IL6) which we'll add later

```{r results = 'hide'}
GSE131411_ensembl <- read_delim("Data/GSE131411_rawcounts.csv", delim = ",") %>% dplyr::rename("Ensembl_ID" = "...1")
```

## Metadata
Load the metadata available with the publication - age, sex, diagnosis but no cell count information or any clinical severity metrics
```{r results = 'hide'}
GSE131411_metadata <- read_delim("Data/GSE131411_metadata.csv",
                                 delim = ",")
```

## Reformat metadata variables
Change some of these variables to factor type (ie an ordinal categorical variable, rather than plain text (character))
```{r results = 'hide'}
GSE131411_metadata$ID <- factor(GSE131411_metadata$ID)
GSE131411_metadata$Sex <- factor(GSE131411_metadata$Sex, levels = c("M", "F"))
GSE131411_metadata$Timepoint <- factor(GSE131411_metadata$Timepoint)
GSE131411_metadata$Timepoint_hr <- factor(GSE131411_metadata$Timepoint_hr, levels = c("16hr", "48hr", "7days"))
GSE131411_metadata$Cohort <- factor(GSE131411_metadata$Cohort)
GSE131411_metadata$D28.mortality <- factor(GSE131411_metadata$D28.mortality)
```

# Process data
## Create biomart object with protein coding genes
This is a linkage table for the different gene nomenclature systems
```{r results = 'hide'}
mart <- useMart("ENSEMBL_MART_ENSEMBL")
mart <- useDataset("hsapiens_gene_ensembl", mart)

# create a lookup matrix using row names
annotLookup <- getBM(
  mart=mart,
  attributes=c("ensembl_gene_id",
    "external_gene_name",
    "entrezgene_id",
    "gene_biotype"),
  filter="ensembl_gene_id",
  values=GSE131411_ensembl$Ensembl_ID,
  uniqueRows=TRUE)

protein_coding_lookup <- annotLookup %>% dplyr::select(external_gene_name, gene_biotype, ensembl_gene_id) %>% dplyr::filter(gene_biotype == "protein_coding" | external_gene_name == "XIST")
```

## Label transcripts with HGNC identifiers and restrict to protein coding genes
Remove duplicated or empty gene IDs
```{r results = 'hide'}
GSE131411_hgnc <- GSE131411_ensembl %>% left_join(., protein_coding_lookup, by = join_by("Ensembl_ID" == "ensembl_gene_id")) %>% select(-gene_biotype, -Ensembl_ID) %>% relocate(external_gene_name) %>% rename(GeneID = external_gene_name) %>% filter(GeneID %in% protein_coding_lookup$external_gene_name)
```

### Filter out duplicated rows
Same geneID, same counts, and then 7 genes with two rows (take the highest count)
```{r}
GSE131411_hgnc <- GSE131411_hgnc %>% filter(GeneID != "") %>% filter(!duplicated(.)) %>% group_by(GeneID) %>% arrange(desc(E07T1)) %>% slice_head(n=1)
```

### Check library size
```{r results = 'hide'}
GSE131411_hgnc %>% column_to_rownames(var = "GeneID") %>% as.matrix() %>% colSums() %>% hist(main = 'Total library sizes across cohort')
```

## Create DGEList objects for raw and CPM values
```{r results = 'hide'}
GSE131411_hgnc_matrix <- GSE131411_hgnc %>% column_to_rownames(var = "GeneID") %>% as.matrix()

GSE131411_hgnc_raw_DGEList <- DGEList(counts=GSE131411_hgnc_matrix,
                                      samples = GSE131411_metadata) 

GSE131411_hgnc_cpm_DGEList <- cpm(GSE131411_hgnc_raw_DGEList)
GSE131411_hgnc_log2cpm_DGEList <- cpm(GSE131411_hgnc_raw_DGEList, log = 2)
```

### Save for CIBERSORTx
Filter for genes occuring with more than 10 average counts in more than 10 samples
This reduces 19k genes to 12k
Save as .tsv for input to CIBERSORTx wep app
```{r results = 'hide'}
CIBERSORT_threshold <- apply(GSE131411_hgnc_matrix, 1, function(x) sum(x > 10) >= 10)
GSE131411_hgnc_CIBERSORT_filtered <- GSE131411_hgnc_matrix[CIBERSORT_threshold, ]
GSE131411_hgnc_CIBERSORT_filtered <- GSE131411_hgnc_CIBERSORT_filtered %>% as.data.frame() %>% rownames_to_column(., var = "Gene")

write.table(GSE131411_hgnc_CIBERSORT_filtered, "CIBERSORTx/GSE131411_CSx_counts.tsv",quote=FALSE, sep='\t', row.names = FALSE)
```

## SepstratifieR
```{r}
# Define gene lists
davenport <- c("ARL14EP","CCNB1IP1","DYRK2","ADGRE3","MDC1","TDRD9","ZAP70")
extended <- c("ARL14EP","CCNB1IP1","DYRK2","ADGRE3","MDC1","TDRD9","ZAP70","DNAJA3","NAT10","THOC1","MRPS9","PGS1","UBAP1","USP5","TTC3","SH3GLB1","BMS1","FBXO31", "SLC25A38")
# define ensembl.id lists
extended.ensembl <- c("ENSG00000144659", "ENSG00000103423", "ENSG00000135372", "ENSG00000079134", "ENSG00000135972", "ENSG00000087157", "ENSG00000165006", "ENSG00000111667", "ENSG00000182670", "ENSG00000097033", "ENSG00000165733", "ENSG00000103264", "ENSG00000152219", "ENSG00000100814", "ENSG00000127334", "ENSG00000131355", "ENSG00000137337", "ENSG00000156414", "ENSG00000115085")

# Change count matrix to correct format - add Ensembl IDs as rownames, then transpose
GSE131411_ensembl_transposed <- GSE131411_ensembl %>% column_to_rownames(var = 'Ensembl_ID') %>% t()

# Run sepstratifieR
predictions.davenport <- stratifyPatients(GSE131411_ensembl_transposed, gene_set = "davenport", k = 20)
predictions.extended <- stratifyPatients(GSE131411_ensembl_transposed, gene_set = "extended", k = 20)
```

### Add SRS predictions to metadata
```{r results = 'hide'}
# Collect predictions
SRS.predictions <- cbind(predictions.davenport@SRS, predictions.davenport@SRSq,predictions.extended@SRS, predictions.extended@SRSq)
colnames(SRS.predictions) <- c("SRS.davenport", "SRSq.davenport","SRS.extended", "SRSq.extended")
SRS.predictions <- SRS.predictions %>% as.data.frame() %>% rownames_to_column(var = 'Sample')

# Convert SRS to factor
SRS.predictions$SRS.davenport <- factor(SRS.predictions$SRS.davenport)
SRS.predictions$SRS.extended <- factor(SRS.predictions$SRS.extended)

# Add to metadata
GSE131411_metadata <- left_join(GSE131411_metadata, SRS.predictions, by = "Sample")
```

## Load in CIBERSORTx data
I've estimated the proportion of different celltypes with the CIBERSORTx web application using the single cell reference derived from Andrew Kwok's paper (https://www.nature.com/articles/s41590-023-01490-5)
Can share these when you need them alter on

Create synthetic variables for total neutrophil, total lymphocyte and NLR
```{r results = 'hide'}
GSE131411_CSx <- read.delim("CIBERSORTx/GSE131411_CSx_results_nodegranulating.txt", stringsAsFactors = FALSE, header= TRUE, sep = "\t") %>% rename_with(~ str_remove_all(., "\\."), everything()) %>% rename(NK_cells = NKcells)

colnames(GSE131411_CSx)

# Calculate total neut and total lymph proportions
GSE131411_CSx <- GSE131411_CSx %>% mutate(total_neut = 
                                            Mature_neutrophils +                
                                            IL1R2_immature_neutrophils + 
                                            S100A89_hi_neutrophils + 
                                            PADI4_immature_neutrophils +
                                            MPO_immature_neutrophils_or_progenitors + 
                                            Cycling_neutrophil_progenitors,
                                          total_lymph = Naive_CD4_T_cells + 
                                            Cycling_TNK + 
                                            Memory_CD4_T_cells + 
                                            Naive_CD8_T_cells +
                                            NK_cells +
                                            CD8_T_cells) %>%
                                  mutate(NLR = total_neut/total_lymph)

# Add all CSx data to main metadata object
GSE131411_metadata <- left_join(GSE131411_metadata, GSE131411_CSx, by = join_by("Sample"=="Mixture"))

```

# Cohort description
```{r}
single_timepoint_metadata <- GSE131411_metadata %>%
  group_by(ID) %>%
  dplyr::filter(row_number() == 1) %>%
  ungroup()

labels <- list(
    variables=list(Sex="Sex",
                   Age="Age", 
                   D28.mortality="Deceased at D28",
                   SRS.extended="Baseline SRS endotype (extended gene set)",
                   SRSq.extended="Baseline SRSq score (extended gene set)"),
    groups=list("Shock cohort", "", ""))

strata <- c(split(single_timepoint_metadata, single_timepoint_metadata$Cohort),list(Total=single_timepoint_metadata))

patient.stratification.arms <- table1(strata, labels, groupspan=c(3,1,1), topclass="Rtable1-zebra", caption = "Baseline characteristics shock cohorts")

patient.stratification.arms
```

# QC
## Check expressed sex vs clinical sex
Often in RNA-seq datasets there are mixups between samples. An easy way to catch most of these is to check whether the clinical sex variable matches the sex of the RNA-seq data (as determined by the expression of X and Y chromosome-specific genes)
It looks like there are no mixups in this dataset which is good. 
```{r results = 'hide'}
# Filter rows for sex-defining genes, then transpose for ggplot, then add metadata
sex_genes <- t(dplyr::filter(as.data.frame(GSE131411_hgnc_log2cpm_DGEList), rownames(GSE131411_hgnc_log2cpm_DGEList) %in% c("XIST", "UTY")))

sex_genes <- sex_genes %>% as.data.frame() %>% rownames_to_column(var = 'Sample') %>% left_join(GSE131411_metadata, ., by = join_by("Sample" == "Sample"))

# Create new variable with predicted sex
sex_genes <- dplyr::mutate(sex_genes, Sex.predicted = case_when(
  XIST > 2 & UTY <2 ~ "F",
  XIST < 2 & UTY >2 ~ "M", 
  XIST > 2 & UTY >2 ~ "Indeterminate"
))

sex_genes <-  dplyr::mutate(sex_genes,
                            Sex.discordance = ifelse(Sex == Sex.predicted, "Match", "Discordant"))
# Convert to factor
variables <- c("Sex.predicted", "Sex.discordance")
sex_genes[variables] <- lapply(sex_genes[variables], factor)
```

## Plot clinical vs genetic sex
```{r results = 'hide'}
ggplot(data=as.data.frame(sex_genes), aes(x=XIST, y=UTY, color = Sex, label = Sample, shape = Cohort)) +
  geom_point() +
  geom_label_repel(data = dplyr::filter(sex_genes, sex_genes$Sex.discordance == "Discordant"), aes(label = Sample)) + 
  labs(x="XIST expression (log2cpm) ", y = "UTY expression (log2cpm)", title = "Sex-defining gene biplot (GSE131411)", color = "Sex (clinical database)") +
  theme_minimal() 
ggsave("Plots/XIST_UTY_sex.png")
```

# SRS analysis
SRS across timepoints, coloured by d28 mortality, faceted by shock group

## Septic shock patients have higher SRSq
```{r results = 'hide'}
GSE131411_metadata %>% ggplot(aes(x=Timepoint_hr, y = SRSq.extended)) + geom_boxplot() +
  facet_wrap(~Cohort) +
  labs(x = 'Timepoint', y = 'SRS (19 gene set)', title = 'SRSq values across study timepoints')
```

## Categorical SRS vs D28 mortality
```{r}
GSE131411_metadata %>% 
  filter(Timepoint == 1) %>%
  ggplot(aes(x = SRS.extended, fill = D28.mortality)) +
  geom_bar(stat = 'count', position = 'fill') +
  facet_wrap(~Cohort)

GSE131411_metadata %>% 
  filter(Timepoint == 1) %>%
  ggplot(aes(x = SRS.extended, fill = D28.mortality)) +
  geom_bar(stat = 'count', position = 'dodge') +
  facet_wrap(~Cohort)
```

## Quantitative SRS (SRSq) vs mortality
```{r results = 'hide'}
GSE131411_metadata %>% ggplot(aes(x=Timepoint_hr, y = SRSq.extended, fill = D28.mortality)) + geom_boxplot() + facet_wrap(~Cohort)
```

```{r results = 'hide'}
GSE131411_metadata %>% ggplot(aes(x=Timepoint_hr, y = SRSq.extended, color = D28.mortality)) +
  geom_point() +
  geom_path(aes(group = 1)) +
  facet_wrap(~Cohort)
```

SRSq versus IL1R2+ neutrophil proportion - expect to see a strong correlation with SRS1 / higher SRSq
This is clear in SS but not in CS
```{r results = 'hide'}
GSE131411_metadata %>% ggplot(aes(x=SRS.extended, y = IL1R2_immature_neutrophils, fill = Cohort)) + geom_boxplot() + facet_wrap(~Cohort) +
  labs(title = 'Categorical SRS versus estimated IL1R2+ neutrophil proportion', x = 'Categorical SRS', y = 'Estimated IL1R2+ neutrophil proportion \n(% of all leukocytes)') +
  theme_minimal()
```


```{r results = 'hide'}
GSE131411_metadata %>% ggplot(aes(x=SRSq.extended, y = IL1R2_immature_neutrophils, fill = Cohort)) + geom_point() + facet_wrap(~Cohort) + labs(title = 'Quantitative SRS versus estimated IL1R2+ neutrophil proportion', x = 'SRSq', y = 'Estimated IL1R2+ neutrophil proportion \n(% of all leukocytes)') + theme_minimal()

```

# CIBERSORTx plots
Melt CIBERSORTx metadata for plotting using facets
```{r results = 'hide'}
melted_cibersort <- GSE131411_metadata %>% reshape2::melt(id.vars = c("Sample", "ID", "Timepoint", "Timepoint_hr", "Cohort", "Sex", "D28.mortality", "Age", "SRS.davenport", "SRSq.davenport", "SRS.extended", "SRSq.extended", "Cause"))
```

Check proportion of cells in mixture
```{r results = 'hide'}
melted_cibersort %>% 
  filter(!variable %in% c("Correlation", "RMSE", "P.value", "NLR", "total_neut", "total_lymph")) %>%
  ggplot(aes(x=reorder(variable,value), y = value)) +
  geom_boxplot(outlier.size = 0.25) +
  coord_flip() +
  theme_minimal() +
  labs(y = "Proportion of all cells in mixture", x = "", title = "Proportion of cells in mixture") +
  theme(axis.text.y=element_text(size=12))
```

Plot estimated NLR
```{r results = 'hide'}
NL_bars <- melted_cibersort %>% 
  filter(variable %in% c("total_neut", "total_lymph")) %>%
  ggplot(aes(x = Timepoint_hr, y = value, fill = variable)) +
  geom_bar(stat = 'identity', position = 'fill') +
  facet_wrap(~Cohort) + 
  theme_void() + 
  labs(y = 'Proportion of leukocytes', title = 'Changes in neutrophil:lymphocyte proportion across timepoints', fill = "") +
  theme(axis.title.y = element_markdown(),
        axis.title.x = element_blank(),
        axis.text.y = element_text(),
        strip.text.x = element_text(size = 10),
        axis.text.x = element_blank(),
        plot.title = element_text(size = 11)) +
        scale_fill_discrete(labels=c('Total Neu.', 'Total Lymph.'))
 # stat_compare_means(method = "anova") 

NL_boxplot <- melted_cibersort %>% 
  filter(variable %in% c("NLR")) %>%
  ggplot(aes(x = Timepoint_hr, y = value)) +
  geom_boxplot() +
  facet_wrap(~Cohort) +
  theme_void() +
  labs(y = "NLR", x = "") +
  theme(axis.title.y = element_text(),
        axis.title.x = element_text(),
        axis.text.y = element_text(),
        strip.text.x = element_blank(),
        axis.text.x = element_text(size = 10)) +
  theme(legend.position = "none")


NL_bars + NL_boxplot + plot_layout(nrow = 2, heights = unit(c(4,4), c('cm', 'null')))
```

Plot NLR across cohorts at each timepoint - clearly very different
```{r results = 'hide'}
GSE131411_metadata %>% ggplot(aes(x=Timepoint_hr, y = NLR, fill = Cohort)) +
  geom_boxplot() +
  ggtitle("Trajectory for predicted NLR across shock groups") +
  theme_minimal() +
  stat_compare_means(method = "t.test")
```

Association of NLR with mortality
D28 mortality probably not the best measure of outcome, but it's all that's available in the public metadata
```{r results = 'hide'}
GSE131411_metadata %>% 
  ggplot(aes(x=D28.mortality, y = NLR, fill = D28.mortality)) + geom_boxplot() +
  ggtitle("Relationship between NLR and D28 mortality across groups") +
  theme_minimal() +
  stat_compare_means(method = "wilcox.test") +
  facet_wrap(~Cohort)
```

# Analysis of estimated cell proportions


Plot relationship between each neutrophil proportion and SRSq across cohorts
Expect to see a relationship between SRSq and IL1R2+, and possibly some other subsets (eg S100A8-high)
```{r results = 'hide'}
melted_cibersort %>%
  filter(!variable %in% c("Correlation", "RMSE", "P.value")) %>%
  filter(grepl("neut", variable)) %>%
ggplot(aes(x=SRSq.extended, y = value, color = Cohort)) +
  geom_point(size = 0.5) +
  geom_smooth() +
  theme_minimal() +
  facet_wrap(~variable)
```

```{r results = 'hide'}
melted_cibersort %>%
  filter(!variable %in% c("Correlation", "RMSE", "P.value")) %>%
  filter(grepl("neut", variable) | variable == "NLR") %>%
ggplot(aes(x=Timepoint, y = value, color = Cohort)) +
  geom_boxplot(size = 0.5) +
  facet_wrap(~variable, scales = 'free')
```

Check for changes in cell proportions across groupings

Colour by SRS group
In SS we see the expected neutrophil proportion differences across SRS groups, these aren't clear in CS. 
```{r results = 'hide'}
melted_cibersort %>% 
  filter(!variable %in% c("Correlation", "RMSE", "P.value", "NLR", "total_neut", "total_lymph")) %>%
ggplot(aes(x=variable, y = value, fill=fct_rev(as.factor(SRS.extended)))) +
  geom_boxplot(outlier.size = 0.25) +
  stat_pwc(method = "t.test", label = "p.adj.signif", hide.ns = TRUE, p.adjust.method = "bonferroni", size = .3, label.size = 80) +
  coord_flip() +
  theme_minimal() +
  labs(y = "Proportion of all cells in mixture", x = "", title = "Abundance of cell types in mixture", fill = "SRS (extended gene set)") +
  guides(fill = guide_legend(reverse=TRUE)) +
  theme(axis.text.y=element_text(size=11)) +
  theme(legend.position="bottom") +
  facet_wrap(~Cohort) 
```

# PCA on gene counts
Perform PCA on the top 2000 most variable genes (as was done in the paper)
Create DESeq2 object
```{r results = 'hide'}
GSE131411_DESeq2 <- DESeqDataSetFromMatrix(countData = round(GSE131411_hgnc_matrix),
                              colData = GSE131411_metadata,
                              design = ~SRS.extended
)
```

Transform into VST-units (variance stabilising transformation)
```{r results = 'hide'}
GSE131411_vst <- vst(GSE131411_DESeq2, blind=FALSE)
```

PCA
Use DESeq function to compute PCs for top 10,000 most variable genes after VST transformation
```{r results = 'hide'}
pcaData10k <- DESeq2::plotPCA(GSE131411_vst, ntop = 2000, intgroup=c("Cohort", "Timepoint", "SRS.extended", "SRSq.extended", "Sex", "IL1R2_immature_neutrophils", "S100A89_hi_neutrophils", "Mature_neutrophils", "NLR", "total_lymph", "total_neut"), returnData=TRUE)
percentVar <- round(100 * attr(pcaData10k, "percentVar"))

```
Additional option: use PCAExplorer for interactive PCA in Shiny app
Can be slow so I've hashed it out here
```{r results = 'hide'}
#pcaExplorer(dds = GSE131411_DESeq2)
```

Plot PC1 vs PC2 coloured by different variables
Summary: PC1 appears to be associated with the NLR, with the study timepoint and with the IL1R2+ proportion. 

```{r results = 'hide'}
pcaData10k %>% ggplot(aes(x=PC1, y = PC2, color = Cohort)) +
  geom_point() +
   xlab(paste0("PC1: ",percentVar[1],"% variance")) +
   ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
   coord_fixed() +
   theme_minimal() +
  ggtitle("Cardiogenic shock vs septic shock")

pcaData10k %>% ggplot(aes(x=PC1, y = PC2, color = NLR)) +
  geom_point() +
   xlab(paste0("PC1: ",percentVar[1],"% variance")) +
   ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
   coord_fixed() +
   theme_minimal() +
  ggtitle("Neutrophil:lymphocyte ratio") +
  labs(subtitle = "Estimated using CIBERSORTx deconvolution") +
  scale_color_viridis() +
  facet_wrap(~Cohort)

pcaData10k %>% ggplot(aes(x=PC1, y = PC2, color = Sex)) +
  geom_point() +
   xlab(paste0("PC1: ",percentVar[1],"% variance")) +
   ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
   coord_fixed() +
   theme_minimal() +
  ggtitle("Male vs female") +
  facet_wrap(~Cohort)

pcaData10k %>% ggplot(aes(x=PC1, y = PC2, color = SRSq.extended)) +
  geom_point() +
   xlab(paste0("PC1: ",percentVar[1],"% variance")) +
   ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
   coord_fixed() +
   theme_minimal() +
  ggtitle("SRSq (extended gene set)") +
  scale_color_viridis() +
  facet_wrap(~Cohort)

pcaData10k %>% ggplot(aes(x=PC1, y = PC2, color = SRS.extended)) +
  geom_point() +
   xlab(paste0("PC1: ",percentVar[1],"% variance")) +
   ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
   coord_fixed() +
   theme_minimal() +
  ggtitle("Categorical SRS (extended gene set)") +
  facet_wrap(~Cohort)

pcaData10k %>% ggplot(aes(x=PC1, y = PC2, color = Timepoint)) +
  geom_point() +
   xlab(paste0("PC1: ",percentVar[1],"% variance")) +
   ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
   coord_fixed() +
   theme_minimal() +
  ggtitle("Timepoint") +
  facet_wrap(~Cohort)

pcaData10k %>% ggplot(aes(x=PC1, y = PC2, color = IL1R2_immature_neutrophils)) +
  geom_point() +
   xlab(paste0("PC1: ",percentVar[1],"% variance")) +
   ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
   coord_fixed() +
   theme_minimal() +
  ggtitle("IL1R2+ neutrophils") +
  scale_color_viridis() +
  facet_wrap(~Cohort)

pcaData10k %>% ggplot(aes(x=PC1, y = PC2, color = S100A89_hi_neutrophils)) +
  geom_point() +
   xlab(paste0("PC1: ",percentVar[1],"% variance")) +
   ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
   coord_fixed() +
   theme_minimal() +
  ggtitle("S100A89_hi_neutrophils") +
  scale_color_viridis() +
  facet_wrap(~Cohort)

pcaData10k %>% ggplot(aes(x=PC1, y = PC2, color = Mature_neutrophils)) +
  geom_point() +
   xlab(paste0("PC1: ",percentVar[1],"% variance")) +
   ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
   coord_fixed() +
   theme_minimal() +
  ggtitle("Mature neutrophils") +
  scale_color_viridis() +
  facet_wrap(~Cohort)

```

# Model counts - neutrophils only
Model changes in the neutrophil subtype proportions over time / between cohorts

Model:
celltype_proportions ~ Cohort + Sex + Timepoint

## Select neutrophil celltypes only
```{r results = 'hide'}
# To make things manageable, extract only the neutrophil celltypes and convert to numerical matrix
GSE131411_metadata_neuts <- GSE131411_metadata %>% select(Sample, contains("neut"), -total_neut) %>% column_to_rownames(var = 'Sample') %>% as.matrix()

# Normalise rows to sum to 1 across neutrophil columns
GSE131411_metadata_neuts <- GSE131411_metadata_neuts/rowSums(GSE131411_metadata_neuts)
```

## Zero replacement
Simplest option - replace missing values with a small constant, here just the smallest non-zero value for each celltype
The neutrophil celltypes have no zero values. 
```{r results = 'hide'}
# Get indices for zero values
GSE131411_metadata_neuts_nonzero <- apply(GSE131411_metadata_neuts, 1, function(x) min(x[x > 0]))

# Replace any zero values with minimum non-zero value for each celltype
for (i in 1:nrow(GSE131411_metadata_neuts)) {
  GSE131411_metadata_neuts[i, GSE131411_metadata_neuts[i, ] == 0] <- GSE131411_metadata_neuts_nonzero[i]
}

# Re-normalise rows to 1
GSE131411_metadata_neuts <- t(t(GSE131411_metadata_neuts) / rowSums(GSE131411_metadata_neuts) * 1)

# Add metadata annotations
GSE131411_metadata_neuts_annotated <- GSE131411_metadata_neuts %>% as.data.frame() %>% rownames_to_column(var = "Sample") %>% left_join(., select(GSE131411_metadata, !any_of(colnames(GSE131411_metadata_neuts))), by = "Sample")                                 
```

## Fit model
```{r results = 'hide'}
library(DirichletReg)

# Create object for use with DirichletReg 
GSE131411_DR_data <- DR_data(Y = GSE131411_metadata_neuts)

# Fit fixed-effects model
DR_model <- DirichReg(formula = GSE131411_DR_data ~ Cohort + Sex + Age + Timepoint_hr, 
                            data = GSE131411_metadata_neuts_annotated,
                            model = 'common')

# Extract fitted values
fitted_values <- fitted(DR_model) %>% as.data.frame()
# Add sample names
fitted_values$Sample <- GSE131411_metadata_neuts_annotated$Sample

# Add metadata back to fitted values
fitted_values <- fitted_values %>% relocate(Sample) %>% left_join(., dplyr::select(GSE131411_metadata_neuts_annotated, Sample, Cohort, Sex, Age, Timepoint, Timepoint_hr, NLR, total_neut), by = "Sample")
```

## Extract model coefficients
```{r}
summary(DR_model)

# Extract the model coefficients and p values
DR_summary <- summary(DR_model)
DR_coef.mat <- DR_summary$coef.mat

# Combine the coefficients matrix and the list of p values. Rename p value column and apply p value adjustment. Add binary variables for sig/non-sig terms and direction of effect. 
DR_coefs_summary <- DR_summary$coefficients %>% as.data.frame() %>% rename(Coefficient = ".") %>% cbind(., DR_coef.mat) %>% rownames_to_column(var = "Term") %>% separate(Term, into = c("Fraction", "Term"), sep = ":", extra = "drop") %>% rename(p = `Pr(>|z|)`) %>% mutate(p.adj = p.adjust(p = p)) %>% mutate(p.adj.sig = ifelse(p.adj <0.05,'sig','ns')) %>%
  mutate(dir_effect = case_when(Coefficient >0 & p.adj.sig == 'sig' ~ "Up",
                                Coefficient <0 & p.adj.sig == 'sig' ~ "Down",
                                .default = NA)) %>% 
  mutate(Term_labeled = case_when(Term == "CohortSS" ~ "Septic shock",
                                  Term == "SexF" ~ "Female sex",
                                  Term == "Age" ~ "Older age",
                                  Term == "Timepoint_hr48hr" ~ "48 hour sample",
                                  Term == "Timepoint_hr7days" ~ "7 day sample",
                                  .default = NA))
```

## Plot significance matrix
This plot adapts the results of summary(DR_model) to show which model terms are associated with each celltype. My interpretation is that the majority of changes are across timepoint (particularly day 7 vs day 1, but also day 2 vs day 1), with only the IL1R2+ count differing between cohorts. 

Simpler version:
```{r results = 'hide'}
ggplot(DR_coefs_summary, aes(x=Fraction, y = Term, fill = -log10(p.adj))) +
  geom_tile() +
  geom_shape() +
  scale_fill_viridis() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  labs(x = "", y = "Model term", fill = "-log10 adjusted p value", title = "Associations with change in neutrophil subtype proportions")
  
```

More complex version showing direction of significant effects
```{r}
  ggplot(DR_coefs_summary, aes(x=Fraction, y = Term, fill = -log10(p.adj))) +
  geom_tile(aes(fill = as.numeric(p.adj))) +
  scale_fill_gradient(trans = "log10",
                      breaks = c(-Inf, 0.001, 0.005, .01, 0.05,Inf),
                      high = 'transparent',
                      #limits=c(0, 0.05),
                      ) +
  geom_tile(aes(color=factor(p.adj.sig, c("sig", "ns"))), fill = '#00000000', size = 0.5) + 
  scale_color_manual(name = "Status after BH adjustment", values = c("red", '#00000000'), drop = FALSE) +
  geom_point(aes(x=Fraction, y=Term, shape = dir_effect), fill="white", col="black", size=1.5) + 
	scale_shape_manual(values=c(25,24),
	                   #labels = c('Increase','Decrease'),
	                   na.translate=FALSE,
	                   drop=FALSE) +
    theme_minimal() +
    theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 10),
          axis.text.y = element_text(size=10)) +
    labs(x = "", y = "Model term", fill = "-log10 adjusted p value", title = "Associations with change in neutrophil subtype proportions", shape = "Direction of significant effect")
```


## Plot proportions by cohort
```{r results = 'hide'}
cohort.labs <- c("Cardiogenic shock", "Septic shock")
names(cohort.labs) <- c("CS", "SS")

fitted_values %>% reshape2::melt(id.vars = c("Sample", "Sex", "Timepoint", "Timepoint_hr", "Age", "Cohort", "total_neut", "NLR")) %>%
  mutate(variable = gsub(".CSx","", variable)) %>%
  mutate(variable = gsub("_"," ", variable)) %>%
  ggplot(aes(x=as.factor(Timepoint_hr), y = value, fill = reorder(variable, value), color = reorder(variable, value))) + 
  geom_bar(position="fill", stat="identity") +
  # geom_text(aes(label = percent(value, accuracy = .1)), 
  #           position = position_fill(vjust = 5), 
  #           colour = 'black') +
  facet_wrap(~Cohort,
             labeller = labeller(Cohort = cohort.labs)) +
  labs(title = 'Shift of neutrophil subtype proportions with time',
       fill = "",
       y = "Proportion of all neutrophils",
       x = "Study timepoint") +
  theme_bw() +
  theme(axis.text = element_text(size=10),
        strip.text.x = element_text(size=10)) +
  guides(color = F)
```

```{r results = 'hide'}
cohort_bar <- fitted_values %>% reshape2::melt(id.vars = c("Sample", "Sex", "Timepoint","Timepoint_hr", "Age", "Cohort", "total_neut", "NLR")) %>%
  mutate(variable = gsub(".CSx","", variable)) %>%
  mutate(variable = gsub("_"," ", variable)) %>%
  ggplot(aes(x=as.factor(Timepoint_hr), y = value, fill = reorder(variable, value), color = reorder(variable, value))) + 
  geom_bar(position="fill", stat="identity") +
  facet_wrap(~Cohort,
             labeller = labeller(Cohort = cohort.labs)) +
  labs(title = 'Shift of neutrophil subtype proportions with time',
       fill = "",
       y = "Proportion of all neutrophils",
       x = "") +
  theme_bw() +
  theme(axis.text = element_blank(),
        strip.text.x = element_text(size=10),
        axis.text.y = element_text()) +
  guides(color = F)

cohort_hist <- fitted_values %>% reshape2::melt(id.vars = c("Sample", "Sex", "Timepoint","Timepoint_hr", "Age", "Cohort", "total_neut", "NLR")) %>%
  mutate(variable = gsub(".CSx","", variable)) %>%
  mutate(variable = gsub("_"," ", variable)) %>%
  ggplot(aes(x=Timepoint_hr, y = total_neut)) +
  geom_boxplot() +
    facet_wrap(~Cohort) +
  theme_void() +
  labs(y = "Neuts <br> % of all leukocytes",
       x = "Study day") +
  theme(axis.title.y = element_markdown(),
        axis.title.x = element_text(),
        axis.text.y = element_text(),
        strip.text.x = element_blank(),
        axis.text.x = element_text(size = 10)) +
 # stat_compare_means(method = "anova") +
  theme(legend.position = "none")

cohort_bar + cohort_hist + plot_layout(nrow = 2, heights = unit(c(6,3), c('cm', 'null')))
```

# Differential expression & over-representation
## Read in msigDB object
```{r results = 'hide'}
msigdbr <- msigdbr(species = "Homo sapiens") %>% mutate(gs_name = gsub("_", " ", gs_name, fixed = TRUE))

# All msigDB sets - exclude vaccine response terms
msigdbr_t2g_all_hgnc <- msigdbr %>% 
  dplyr::distinct(gs_name, human_gene_symbol) %>% 
  as.data.frame()

# Filter to hallmark sets
msigdbr_t2g_hallmarks_hgnc <-msigdbr %>% 
  dplyr::filter(gs_cat == "H") %>% 
  dplyr::distinct(gs_name, human_gene_symbol) %>%
  as.data.frame() %>%
  mutate(gs_name = gsub("HALLMARK", "", gs_name))

# Celltype sets only
msigdbr_t2g_celltypes_hgnc <-msigdbr %>% 
  dplyr::filter(gs_cat == "C8") %>% 
  dplyr::distinct(gs_name, human_gene_symbol) %>% 
  as.data.frame()

# C2 sets only, remove investigator names
msigdbr_t2g_C2_hgnc <-msigdbr %>% 
  dplyr::filter(gs_cat == "C2") %>% 
  dplyr::distinct(gs_name, human_gene_symbol) %>% 
  as.data.frame() %>%
  mutate(gs_name = sub(".*? ", "", gs_name))
  
```

## Create gene background object
*Subset of genes with >10 average raw counts in experiment*
```{r}
# Create background geneset using all genes with >10 average raw read count
thresholded.background <- GSE131411_hgnc_matrix[which(rowMeans(GSE131411_hgnc_matrix)>=10),]

# Add ENSEMBL and ENTREZ IDs
thresholded.background <- bitr(rownames(thresholded.background), 
            fromType = "SYMBOL", 
            toType = c("ENSEMBL", "ENTREZID"), 
            OrgDb = "org.Hs.eg.db")

# Check for duplicate ENSEMBL ID values and filter out
thresholded.background$duplicate.ensembl <- stri_duplicated(thresholded.background$ENSEMBL)

deduplicated.thresholded.background <- dplyr::filter(thresholded.background, thresholded.background$duplicate.ensembl==FALSE)

# Convert Ensembl ID column to character to use as final background list
background_ensembl <- as.character(deduplicated.thresholded.background$ENSEMBL)
background_hgnc <- as.character(deduplicated.thresholded.background$SYMBOL)

length(background_ensembl)
```

## Mixed effect model - no celltypes
This model replicates what they did in the paper, looking for genes which are differentially expressed across each of:
1. Patient sex
2. Patient age
3. Cohort (SS vs CS)

accounting for the multiple timepoints using a linear mixed model

but without accounting for any baseline differences in celltype values. 

Primary model: transcript count ~ Sex + Age + Cohort + Timepoint + (1|ID)
```{r warning=FALSE}
# Move metadata sample names to rownames for lmer
metadata_lmer <- GSE131411_metadata %>% column_to_rownames(var = "Sample") %>% dplyr::select(Sex, Age, Cohort, Timepoint, ID)

deseq_lmer <- GSE131411_vst@assays@data[[1]] %>% as.data.frame()

formula <- '~ Cohort*Timepoint + (1|ID)'

# Run lmerSeq with mixed effects model
fit.lmerSeq <- lmerSeq.fit(form = ~ Cohort + Age + Sex + Timepoint + (1|ID),
                            expr_mat = deseq_lmer,
                            sample_data = metadata_lmer,
                            REML = T)

# Print model matrix - use this to select the contrast to extract coefficients for and to plot
model.matrix(fit.lmerSeq[[1]]$fit) %>% as.data.frame()
```

### Time 1 vs time 3
Extract the genes showing a significant difference in expression level between timepoint 1 (16 hrs) and timepoint 3 (7 days)
As can be seen there are over 3000 genes differentially expressed
```{r results = 'hide'}
contrasts=rbind(earlyvlate = c(0,0,0,0,0,1))

contrast_summary <- lmerSeq.contrast(lmerSeq_results = fit.lmerSeq,
                                       contrast_mat = contrasts,
                                       p_adj_method = 'BH',
                                       ddf = 'Satterthwaite',
                                       sort_results = T)

contrast_summary$summary_table$diffexpressed <- ifelse(contrast_summary$summary_table$p_val_adj < 0.05 & (contrast_summary$summary_table$Estimate <0.6 | contrast_summary$summary_table$Estimate >0.6), "Sig.", "Non-sig.")
contrast_summary$summary_table$diffexpressed <- factor(contrast_summary$summary_table$diffexpressed, levels = c('Sig.', 'Non-sig.'))

contrast_summary$summary_table$delabel <- ifelse(contrast_summary$summary_table$diffexpressed == "Sig.", contrast_summary$summary_table$gene , NA)

table(contrast_summary$summary_table$p_val_adj < 0.05)

genes_up <- contrast_summary$summary_table %>% filter(p_val_adj < 0.05 & Estimate >0 ) %>% dplyr::select(gene) %>% as.list()
genes_down <- contrast_summary$summary_table %>% filter(p_val_adj < 0.05 & Estimate <0 ) %>% dplyr::select(gene) %>% as.vector()

volcano <- ggplot(data=contrast_summary$summary_table, aes(x=Estimate, y=-log10(p_val_adj), col=diffexpressed, label=delabel)) +
        geom_point(size=.5) + 
        theme_minimal() +
        geom_text_repel(max.overlaps = 14, colour='black', size = 3) +
        geom_vline(xintercept=c(-0.6, 0.6)) +
        geom_hline(yintercept=-log10(0.05)) +
        labs(title = "Differentially expressed genes: Earliest vs day 7 samples",
          color = 'Differentially expressed',
             x = 'Model estimate (transformed expression units)',
             y = '-log10(p. adjust)',
             caption = paste0("Model design: ", formula, "\n", 
                              "Genes up: ", length(genes_up$gene), " Genes down: ", length(genes_down$gene), " Total genes: ", nrow(contrast_summary$summary_table)))
volcano

# Over-representation test
msig_enriched_up <- enricher(gene = genes_up$gene, 
                            TERM2GENE = msigdbr_t2g_hallmarks_hgnc, 
                            universe = background_hgnc,
                            pvalueCutoff = .05,
                            qvalueCutoff = .05)

msig_enriched_up <- msig_enriched_up %>% mutate(direction = 'up') %>% as_tibble()

msig_enriched_down <- enricher(gene = genes_down$gene, 
                            TERM2GENE = msigdbr_t2g_hallmarks_hgnc, 
                            universe = background_hgnc,
                            pvalueCutoff = .05,
                            qvalueCutoff = .05)

msig_enriched_down <- msig_enriched_down %>% mutate(direction = 'down') %>% as_tibble()

# Order labels
msig_enriched_up$ID <- factor(msig_enriched_up$ID, levels = rev(msig_enriched_up$ID))

dotplot_up <- msig_enriched_up %>% slice_head(n=10) %>%
  ggplot(aes(x = -log10(p.adjust), y=ID, size = Count, fill = -log10(p.adjust))) +
  geom_point(shape = 21) +
  scale_y_discrete(labels=function(x) str_wrap(x, width=30)) +
  labs(y = "",
       title = "Gene sets with increased expression at\n 7 days compared to 16 hours",
       subtitle = "")
dotplot_up

msig_enriched_down$ID <- factor(msig_enriched_down$ID, levels = rev(msig_enriched_down$ID))

dotplot_down <- msig_enriched_down %>% slice_head(n=10) %>%
  ggplot(aes(x = -log10(p.adjust), y=ID, size = Count, fill = -log10(p.adjust))) +
  geom_point(shape = 21) +
  scale_y_discrete(labels=function(x) str_wrap(x, width=30)) +
  labs(y = "",
       title = "Gene sets with decreased expression at\n 7 days compared to 16 hours",
       subtitle = "")
dotplot_down
```

### Time 1 vs time 2
Extract the genes showing a significant difference in expression level between timepoint 1 (16 hrs) and timepoint 3 (7 days)
```{r results = 'hide'}
contrasts=rbind(earlyvlate = c(0,0,0,0,1,0))

contrast_summary <- lmerSeq.contrast(lmerSeq_results = fit.lmerSeq,
                                       contrast_mat = contrasts,
                                       p_adj_method = 'BH',
                                       ddf = 'Satterthwaite',
                                       sort_results = T)

contrast_summary$summary_table$diffexpressed <- ifelse(contrast_summary$summary_table$p_val_adj < 0.05 & (contrast_summary$summary_table$Estimate <0.6 | contrast_summary$summary_table$Estimate >0.6), "Sig.", "Non-sig.")
contrast_summary$summary_table$diffexpressed <- factor(contrast_summary$summary_table$diffexpressed, levels = c('Sig.', 'Non-sig.'))

contrast_summary$summary_table$delabel <- ifelse(contrast_summary$summary_table$diffexpressed == "Sig.", contrast_summary$summary_table$gene , NA)

table(contrast_summary$summary_table$p_val_adj < 0.05)

genes_up <- contrast_summary$summary_table %>% filter(p_val_adj < 0.05 & Estimate >0 ) %>% dplyr::select(gene) %>% as.list()
genes_down <- contrast_summary$summary_table %>% filter(p_val_adj < 0.05 & Estimate <0 ) %>% dplyr::select(gene) %>% as.vector()

volcano <- ggplot(data=contrast_summary$summary_table, aes(x=Estimate, y=-log10(p_val_adj), col=diffexpressed, label=delabel)) +
        geom_point(size=.5) + 
        theme_minimal() +
        geom_text_repel(max.overlaps = 14, colour='black', size = 3) +
        geom_vline(xintercept=c(-0.6, 0.6)) +
        geom_hline(yintercept=-log10(0.05)) +
        labs(title = "Differentially expressed genes: Earliest vs 48hr samples",
          color = 'Differentially expressed',
             x = 'Model estimate (transformed expression units)',
             y = '-log10(p. adjust)',
             caption = paste0("Model design: ", formula, "\n", 
                              "Genes up: ", length(genes_up$gene), " Genes down: ", length(genes_down$gene), " Total genes: ", nrow(contrast_summary$summary_table)))
volcano
ggsave("Plots/DEG_early_vs_middle.png", width = 3, height = 3)

# Over-representation test
msig_enriched_up <- enricher(gene = genes_up$gene, 
                            TERM2GENE = msigdbr_t2g_hallmarks_hgnc, 
                            universe = background_hgnc,
                            pvalueCutoff = .05,
                            qvalueCutoff = .05)

msig_enriched_up <- msig_enriched_up %>% as_tibble()

msig_enriched_down <- enricher(gene = genes_down$gene, 
                            TERM2GENE = msigdbr_t2g_hallmarks_hgnc, 
                            universe = background_hgnc,
                            pvalueCutoff = .05,
                            qvalueCutoff = .05)

msig_enriched_down <- msig_enriched_down %>% as_tibble()

# Order labels
msig_enriched_up$ID <- factor(msig_enriched_up$ID, levels = rev(msig_enriched_up$ID))

dotplot_up <- msig_enriched_up %>% slice_head(n=10) %>%
  ggplot(aes(x = -log10(p.adjust), y=ID, size = Count, fill = -log10(p.adjust))) +
  geom_point(shape = 21) +
  scale_y_discrete(labels=function(x) str_wrap(x, width=30)) +
  labs(y = "",
       title = "Gene sets with increased expression at\n 48 hours compared to 16 hours",
       subtitle = "")
dotplot_up

msig_enriched_down$ID <- factor(msig_enriched_down$ID, levels = rev(msig_enriched_down$ID))

dotplot_down <- msig_enriched_down %>% slice_head(n=10) %>%
  ggplot(aes(x = -log10(p.adjust), y=ID, size = Count, fill = -log10(p.adjust))) +
  geom_point(shape = 21) +
  scale_y_discrete(labels=function(x) str_wrap(x, width=30)) +
  labs(y = "",
       title = "Gene sets with reduced expression at\n 48 hours compared to 16 hours",
       subtitle = "")
dotplot_down
```

### Cardiac shock vs septic shock
Extract genes which are differentially expressed across cardiogenic shock and septic shock 
As was seen in the paper, there are no genes in this category
```{r results = 'hide'}
contrasts=rbind(earlyvlate = c(0,1,0,0,0,0))

contrast_summary <- lmerSeq.contrast(lmerSeq_results = fit.lmerSeq,
                                       contrast_mat = contrasts,
                                       p_adj_method = 'BH',
                                       ddf = 'Satterthwaite',
                                       sort_results = T)

contrast_summary$summary_table$diffexpressed <- ifelse(contrast_summary$summary_table$p_val_adj < 0.05 & (contrast_summary$summary_table$Estimate <0.6 | contrast_summary$summary_table$Estimate >0.6), "Sig.", "Non-sig.")
contrast_summary$summary_table$diffexpressed <- factor(contrast_summary$summary_table$diffexpressed, levels = c('Sig.', 'Non-sig.'))

contrast_summary$summary_table$delabel <- ifelse(contrast_summary$summary_table$diffexpressed == "Sig.", contrast_summary$summary_table$gene , NA)

table(contrast_summary$summary_table$p_val_adj < 0.05)

genes_up <- contrast_summary$summary_table %>% filter(p_val_adj < 0.05 & Estimate >0 ) %>% dplyr::select(gene) %>% as.list()
genes_down <- contrast_summary$summary_table %>% filter(p_val_adj < 0.05 & Estimate <0 ) %>% dplyr::select(gene) %>% as.vector()

volcano <- ggplot(data=contrast_summary$summary_table, aes(x=Estimate, y=-log10(p_val_adj), col=diffexpressed, label=delabel)) +
        geom_point(size=.5) + 
        theme_minimal() +
        geom_text_repel(max.overlaps = 14, colour='black', size = 3) +
        geom_vline(xintercept=c(-0.6, 0.6)) +
        geom_hline(yintercept=-log10(0.05)) +
        labs(title = "Differentially expressed genes: Cardiac shock vs septic shock samples",
          color = 'Differentially expressed',
             x = 'Model estimate (transformed expression units)',
             y = '-log10(p. adjust)',
             caption = paste0("Model design: ", formula, "\n", 
                              "Genes up: ", length(genes_up$gene), " Genes down: ", length(genes_down$gene), " Total genes: ", nrow(contrast_summary$summary_table)))
volcano
ggsave("Plots/CS_vs_SS.png", width = 3, height = 3)
```

### Male versus female
As a control, extract the genes differentially expressed across sex groups. Expect to see X and Y chromosome genes if there is enough power to detect them. 
```{r results = 'hide'}
contrasts=rbind(earlyvlate = c(0,0,1,0,0,0))

contrast_summary <- lmerSeq.contrast(lmerSeq_results = fit.lmerSeq,
                                       contrast_mat = contrasts,
                                       p_adj_method = 'BH',
                                       ddf = 'Satterthwaite',
                                       sort_results = T)

contrast_summary$summary_table$diffexpressed <- ifelse(contrast_summary$summary_table$p_val_adj < 0.05 & (contrast_summary$summary_table$Estimate <0.6 | contrast_summary$summary_table$Estimate >0.6), "Sig.", "Non-sig.")
contrast_summary$summary_table$diffexpressed <- factor(contrast_summary$summary_table$diffexpressed, levels = c('Sig.', 'Non-sig.'))

contrast_summary$summary_table$delabel <- ifelse(contrast_summary$summary_table$diffexpressed == "Sig.", contrast_summary$summary_table$gene , NA)

table(contrast_summary$summary_table$p_val_adj < 0.05)

genes_up <- contrast_summary$summary_table %>% filter(p_val_adj < 0.05 & Estimate >0 ) %>% dplyr::select(gene) %>% as.list()
genes_down <- contrast_summary$summary_table %>% filter(p_val_adj < 0.05 & Estimate <0 ) %>% dplyr::select(gene) %>% as.vector()

volcano <- ggplot(data=contrast_summary$summary_table, aes(x=Estimate, y=-log10(p_val_adj), col=diffexpressed, label=delabel)) +
        geom_point(size=.5) +
        theme_minimal() +
        geom_text_repel(max.overlaps = 14, colour='black', size = 3) +
        geom_vline(xintercept=c(-0.6, 0.6)) +
        geom_hline(yintercept=-log10(0.05)) +
        labs(title = "Differentially expressed genes: Male versus female patients",
          color = 'Differentially expressed',
             x = 'Model estimate (transformed expression units)',
             y = '-log10(p. adjust)',
             caption = paste0("Model design: ", formula, "\n",
                              "Genes up: ", length(genes_up$gene), " Genes down: ", length(genes_down$gene), " Total genes: ", nrow(contrast_summary$summary_table)))
volcano
```
